src/postcard_generator.py

```diff
@@ -0,0 +1,59 @@
+import speech_recognition as sr
+import streamlit as st
+import torch
+from diffusers import StableDiffusionPipeline
+from PIL import Image
+
+MODEL_ID = "runwayml/stable-diffusion-v1-5"
+IMAGES_DIR = "../images/"
+SYSTEM = "cuda" if torch.cuda.is_available() else "cpu"
+
+
+@st.cache(allow_output_mutation=True, suppress_st_warning=True)
+def get_pipe():
+    if SYSTEM == "cuda":
+        return StableDiffusionPipeline.from_pretrained(MODEL_ID, torch_dtype=torch.float16, revision="fp16").to(SYSTEM)
+    else:
+        return StableDiffusionPipeline.from_pretrained(MODEL_ID).to(SYSTEM)
+
+
+def model_image(txt, theme, steps, height, width):
+    pipe = get_pipe()
+    pipe.safety_checker = lambda images, clip_input: (images, False)
+    prompt = f"Cartoon {theme} postcard {txt}"
+    image = pipe(prompt=prompt, num_inference_steps=steps, height=height, width=width).images[0]
+    return image
+
+
+def record_voice():
+    try:
+        listener = sr.Recognizer()
+        with sr.Microphone() as microphone:
+            aud_msg = listener.listen(microphone)
+        txt_msg = listener.recognize_google(aud_msg, language='en-US')
```

Comment:
Btw, you can find ASR models on HugginFace. So you can combine 2 neural networks in your app 😍
https://huggingface.co/models?pipeline_tag=automatic-speech-recognition

Reply:
Done!